hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotManager.java

```diff
@@ -157,8 +162,23 @@ public static DBCheckpoint createOmSnapshotCheckpoint(
       throws IOException {
     RDBStore store = (RDBStore) omMetadataManager.getStore();
 
-    final DBCheckpoint dbCheckpoint = store.getSnapshot(
-        snapshotInfo.getCheckpointDirName());
+    final DBCheckpoint dbCheckpoint;
+
+    // Acquire deletedTable write lock
+    omMetadataManager.getTableLock(OmMetadataManagerImpl.DELETED_TABLE)
```

Comment:
This seems to be the only place we are taking the lock ? Shouldn't we be taking the lock in delete code path as well ?

Reply:
DeleteSnapshot request only marks a snapshot as deleted but doesn't do any cleanup. Actual clean up is done in @aswinshakil 's pending PR #4244 . And yes that needs to hold the `deletedTable` write lock when this PR is done.